src/content/app-architecture/case-study/index.md

```diff
@@ -0,0 +1,198 @@
+   would have directories
+   called `repositories`, `models`, `services`, `viewmodels`, etc.
+
+The architecture recommended in this guide lends itself to 
+a combination of the two. 
+Data layer objects (repositories and services) aren't tied to a single feature, 
+while UI layer objects (views and ViewModels) are. 
+The following is how the code is organized within the Compass application.
+
+```text
+lib
+|____ui
+| |____core
+| | |____ui
+| | | |____<shared widgets>
+| | |____themes
+| |____<FEATURE NAME>
+| | |____view_model
+| | | |_____<view_model class>.dart
+| | |____widgets
+| | | |____<feature name>_screen.dart
+| | | |____<other widgets>
+|____domain
+| |____models
+| | |____<model name>.dart
+|____data
+| |____repositories
+| | |____<repository class>.dart
+| |____services
+| | |____<service class>.dart
+| |____model
+| | |____<api model class>.dart
+|____config
+|____utils
+|____routing
+|____main_staging.dart
+|____main_development.dart
+|____main.dart
+
+// The test folder contains unit and widget tests
+test
+|____data
+|____domain
+|____ui
+|____utils
+
+// The testing folder contains mocks other classes need to execute tests
+testing
+|____fakes
+|____models
+```
+
+The important designations are within the `data` and `ui` folders. 
+The data folder organizes code by type, 
+because repositories and services can be used across different features and 
+by multiple ViewModels. The ui folder organizes the code by feature, 
+because each feature will have exactly one view and exactly one ViewModel.
+
+Other notable features of this folder structure:
+
+* The UI folder also contains a subdirectory called "core". Core contains
+  widgets and theme logic that is shared by multiple views, such as buttons with
+  your brand styling.
+* The domain folder contains the application data types, because they're used by
+  the data and ui layers.
+* The app contains three "main" files, which act as different entry points to
+  the application for development, staging and production.
+* There are two test-related directories at the same level as lib: `test/` has
+  the test code, and it's own structure matches `lib/`. `testing/` is a
+  subpackage that contains mocks and other testing utilities which may be used
+  in other packages' test code. The `testing/` folder could be described as a
+  version of your app that you don't ship. It is the content that is tested.
+
+There's additional code in the compass app that doesn't pertain to architecture.
+For the full package structure, [view it on Github][].
+
+## Other architecture options
+
+The example in this case-study demonstrates how one application abides by our
+recommended architectural rules, but there are many other example apps that
+could've been written. The UI of this app leans heavily on `ViewModels`
+and `ChangeNotifier`, but it could've easily been written
+with `Streams`, [Riverpod][], [flutter_bloc][], [Signals][],
```

Comment:
Since there is no type named `Streams`, you can just says "streams". If you want to be clear it's referring to the type, you could say something like "Dart's `Stream` support."